### PR TITLE
added processor and publisher examples along with tests

### PR DESCRIPTION
**Pull request by candysmurf**
#### Summary
* added a fake processor
* added a fake publisher

#### Test done
* small tests
* medium tests

### CODE REVIEW

File: examples/collector/rand/rand_test.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+			So(err, ShouldBeNil)
+			So(mts[0].Data, ShouldEqual, "charm")
+		})
+	})
+
+	Convey("Test GetMetricTypes", t, func() {
+		rc := RandCollector{}
+		cfg := map[string]interface{}{
+			"server": "xyz.com",
+			"port":   12345,
```
Comment (IRCody) [inline examples/collector/rand/rand_test.go]: Is there any reasoning behind putting a non-blank config in? Is the value used in the plugin?

File: examples/processor/reverse/reverse.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+func (r RProcessor) Process(mts []plugin.Metric, cfg plugin.Config) ([]plugin.Metric, error) {
+	metrics := []plugin.Metric{}
+	for _, m := range mts {
+		switch m.Data.(type) {
+		case int:
+			m.Data = stringToInt(reverse(intToString(m.Data.(int))))
+		case int32:
+			i32 := int(m.Data.(int32))
+			m.Data = stringToInt(reverse(intToString(i32)))
+		case int64:
```
Comment (IRCody) [inline examples/processor/reverse/reverse.go]: If we are going to special case all the int types would it make sense to also add the `uint` types?

File: examples/processor/reverse/reverse.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+	policy.AddBoolRule([]string{"random"}, br)
+
+	return *policy, nil
+}
+
+func reverse(s string) string {
+	r := []rune(s)
+	for i, j := 0, len(r)-1; i < len(r)/2; i, j = i+1, j-1 {
+		r[i], r[j] = r[j], r[i]
+	}
```
Comment (IRCody) [inline examples/processor/reverse/reverse.go]: This is a small thing but there is a lot going on in this loop. Would it be more clear as:
```go
l := len(r)
for i := 0; i < len(r)/2; i++ {
    r[i], r[l-i-1] = r[l-i-1], r[i]
}
```

Also, what is the advantage of going from string->rune -> string?

File: examples/publisher/file/file.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+	"bufio"
+	"fmt"
+	"os"
+
+	"github.com/intelsdi-x/snap-plugin-lib-go/v1/plugin"
+)
+
+// FPublisher is a testing publisher.
+type FPublisher struct {
+}
```
Comment (IRCody) [inline examples/publisher/file/file.go]: This bracket should be on the line above for consistency with the other plugin.

File: v1/plugin/plugin_proxy_test.go [hunk 5]
```diff
@@ -22,6 +22,7 @@ limitations under the License.
 package plugin
```
Comment (IRCody) [inline v1/plugin/plugin_proxy_test.go]: This is a good change but I think it should be it's own commit.

File: examples/processor/reverse/reverse.go (hunk 2 above)
Comment (candysmurf) [inline examples/processor/reverse/reverse.go]: it makes sense to add all built-in types. Only that I thought this is a fake one I didn't add all of them.

**Comment by IRCody** [comment]
I think to we could add something like this line at the top of the both plugins to provide compile time error for not conforming to the plugin interface:
```go
var _ plugin.Collector = (*RandCollector)(nil)
```

Replace RandCollector with the other plugin implementations with the appropriate interface.

File: examples/processor/reverse/reverse.go (hunk 2 above)
Comment (IRCody) [inline examples/processor/reverse/reverse.go]: That's a good point. Probably no need to add all of them.

File: examples/processor/reverse/reverse.go (hunk 3 above)
Comment (candysmurf) [inline examples/processor/reverse/reverse.go]: @cody, your code does not run:-) LOL. What I used actually is Andrew Gerrand's code: https://github.com/golang/example/blob/master/stringutil/reverse.go.

From what I read, string stores in []byte. It's better to convert it to rune(int32) for the efficiency.
https://blog.golang.org/strings

Comment (IRCody) [inline examples/processor/reverse/reverse.go]: > @cody, 

My github user name is ircody, not cody. I'm sure @ cody doesn't want to be notified for this.

>your code does not run:-) LOL. What I used actually is Andrew Gerrand's code: https://github.com/golang/example/blob/master/stringutil/reverse.go.

It does run. See [playgroud example](https://play.golang.org/p/RE2-dmL2KC). Having multiple iterators makes it harder to read imo but like I said it's a small thing. Probably should have added that it was just a style comment not functional.

> From what I read, string stores in []byte. It's better to convert it to rune(int32) for the efficiency.
https://blog.golang.org/strings

I didn't see anything regarding efficiency in that post. The reasoning for converting to runes is to support unicode characters easily.

**Comment by IRCody** [comment]
LGTM
